### PR TITLE
Added content about deployment modes

### PR DESCRIPTION
**Pull request by martin-neotech**
Improved deployment modes content.
Dedicated web server documentation.

### CODE REVIEW

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 1]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+**3. Host web application with a HTTP server.**
+
+Start a web server (in this example you will start a web server with Python 3) to serve the content:
+
+[source, shell]
+----
+python -m http.server --directory browser/ 8080
+----
+
+Point your web browser to `http://localhost:8080`.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:180]: Again the link formatting. We have to decide how we format the links and make the documentation consistent.

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 2]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+**4. Test connecting to a Neo4j database instance.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:182]:
```suggestion
. Test the connection to a Neo4j database instance.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 3]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+Start a Neo4j database instance to test for connection.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:184]:
```suggestion

```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 4]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+If you have the Neo4j tarball you can start the database instance with the command (assuming that the CWD is _<neo4j-home>_):
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:185]:
```suggestion
+
If you have the Neo4j tarball, you can start a database instance with the following command (assuming that the CWD is _<neo4j-home>_):
```
What does "CWD" stand for?

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 5]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+[source, shell]
+----
+bin/neo4j start
+----
+
+In Neo4j Browser connect to the Neo4j database instance with the URI:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:192]:
```suggestion
In Neo4j Browser, connect to the Neo4j database instance with the  Neo4j Browser URI:
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 6]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+[source, role=noheader]
+----
+neo4j://localhost:7687
+----
+====
+
+See xref:operations#create-dbms-connection[Create a DBMS connection] for more details how to setup a connection to Neo4j in Neo4j Browser.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:200]:
```suggestion
For details on how to set up a connection to Neo4j in Neo4j Browser, see xref:operations#create-dbms-connection[Create a DBMS connection].
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 7]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+
+=== Build and run web server
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:203]:
```suggestion
[[build-web-server]]
=== Build and run a web server
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 8]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+You can build the source code and run Neo4j Browser.
+
+The source code can be found at link:https://github.com/neo4j/neo4j-browser[https://github.com/neo4j/neo4j-browser].
+
+.Build and run web server
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:209]:
```suggestion
.Build and run a web server
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 9]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+====
+
+**1. Get the source code.**
+
+Clone the Neo4j Browser git repository.
+The command will only get the `master` branch to save bandwidth and file space.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:215]:
```suggestion
The command gets only the `master` branch to save bandwidth and file space.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 10]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+[source, shell]
+----
+git clone --single-branch --branch=master https://github.com/neo4j/neo4j-browser/
+----
+
+Change the current working directory to the root of the git repository.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:222]:
```suggestion
.. Change the current working directory to the root of the Git repository.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 11]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+=== Get pre build and run web server
+
+You can download Neo4j Browser from Maven (link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[`https://search.maven.org/artifact/org.neo4j.client/neo4j-browser`]) and serve the web application with a HTTP server.
+
+
+.Get pre build and run web server
+====
+
+**1. Download Neo4j Browser.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:151]:
```suggestion
. Download Neo4j Browser from Maven.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 12]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+Download Neo4j Browser from Maven:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:153]:
```suggestion

```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 13]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+[source, shell, subs="attributes+"]
+----
+wget https://repo1.maven.org/maven2/org/neo4j/client/neo4j-browser/{neo4j-browser-version}/neo4j-browser-{neo4j-browser-version}.jar
+----
+
+Or manually go to link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[`https://search.maven.org/artifact/org.neo4j.client/neo4j-browser`] and download Neo4j Browser.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:160]:
```suggestion
Alternatively, go to link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[`https://search.maven.org/artifact/org.neo4j.client/neo4j-browser`] and download Neo4j Browser.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 14]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+**2. Unzip the .jar file.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:162]:
```suggestion
. Unzip the .jar file.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 15]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
+Unzip the `.jar` file:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:164]:
```suggestion

```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 16]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+You can download Neo4j Browser from Maven (link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[`https://search.maven.org/artifact/org.neo4j.client/neo4j-browser`]) and serve the web application with a HTTP server.
+
+
+.Get pre build and run web server
+====
+
+**1. Download Neo4j Browser.**
+
+Download Neo4j Browser from Maven:
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:154]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 17]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell, subs="attributes+"]
+----
+wget https://repo1.maven.org/maven2/org/neo4j/client/neo4j-browser/{neo4j-browser-version}/neo4j-browser-{neo4j-browser-version}.jar
+----
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:159]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 18]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Or manually go to link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[`https://search.maven.org/artifact/org.neo4j.client/neo4j-browser`] and download Neo4j Browser.
+
+**2. Unzip the .jar file.**
+
+Unzip the `.jar` file:
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:165]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 19]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell, subs="attributes+"]
+----
+unzip neo4j-browser-{neo4j-browser-version}.jar
+----
+
+**3. Host web application with a HTTP server.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:171]:
```suggestion
. Host a web application with an HTTP server.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 20]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:172]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 21]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Start a web server (in this example you will start a web server with Python 3) to serve the content:
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:174]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 22]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+python -m http.server --directory browser/ 8080
+----
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:179]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc (hunk 1 above)
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:180]:
```suggestion
.. Point your web browser to `http://localhost:8080`.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 23]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Point your web browser to `http://localhost:8080`.
+
+**4. Test connecting to a Neo4j database instance.**
+
+Start a Neo4j database instance to test for connection.
+If you have the Neo4j tarball you can start the database instance with the command (assuming that the CWD is _<neo4j-home>_):
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:186]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 24]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+bin/neo4j start
+----
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:191]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 25]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+In Neo4j Browser connect to the Neo4j database instance with the URI:
+
+[source, role=noheader]
+----
+neo4j://localhost:7687
+----
+====
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:199]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 26]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+=== Build and run web server
+
+You can build the source code and run Neo4j Browser.
+
+The source code can be found at link:https://github.com/neo4j/neo4j-browser[https://github.com/neo4j/neo4j-browser].
+
+.Build and run web server
+====
+
+**1. Get the source code.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:212]:
```suggestion
. Get the source code.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 27]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:213]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 28]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Clone the Neo4j Browser git repository.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:214]:
```suggestion
.. Clone the Neo4j Browser Git repository. +
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 29]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+The command will only get the `master` branch to save bandwidth and file space.
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:216]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 30]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+git clone --single-branch --branch=master https://github.com/neo4j/neo4j-browser/
+----
+
+Change the current working directory to the root of the git repository.
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:223]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 31]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+cd neo4j-browser
+----
+
+**2. Build with Node.js**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:229]:
```suggestion
. Build with Node.js.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 32]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:230]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 33]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Install Node.js dependencies:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:231]:
```suggestion
.. Install Node.js dependencies:
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 34]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:232]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 35]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+npm install
+----
+
+Build the project:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:238]:
```suggestion
.. Build the project.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 36]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:239]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 37]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+npm build
+----
+
+**3. Host web application with a HTTP server.**
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:245]:
```suggestion
. Host web application with an HTTP server.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 38]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:246]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 39]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Start a web server (in this example you will start a web server with Python 3) to serve the content:
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:247]:
```suggestion
.. Start a web server to serve the content.
In this example, you start a web server with Python 3.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 40]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:248]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 41]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+[source, shell]
+----
+python -m http.server --directory dist/ 8080
+----
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:253]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 42]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+Point your web browser to `http://localhost:8080`.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:254]:
```suggestion
.. Point your web browser to `http://localhost:8080`.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 43]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:255]:
```suggestion
+
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 44]
```diff
@@ -52,36 +65,194 @@ If you forget it, you have to remove the credentials file in _data/dbms/authoriz
...
+See xref:operations#create-dbms-connection[Create a DBMS connection] for details how to setup a connection to Neo4j in Neo4j Browser.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:256]:
```suggestion
For details on how to set up a connection to Neo4j in Neo4j Browser, see xref:operations#create-dbms-connection[Create a DBMS connection].
```

**Review by renetapopova — CHANGES_REQUESTED** [review]
Thanks @martin-neotech! Really valuable content! See my suggestions inline.

File: docs/modules/ROOT/pages/deployment-modes.adoc (hunk 4 above)
Comment (martin-neotech) [reply docs/modules/ROOT/pages/deployment-modes.adoc:185]: current working directory

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 45]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+This case is valid if you have access to a Neo4j instance and dont want to install Neo4j Desktop or dont want to use a hosted Neo4j Browser web application.
+It is a simple sollution, start a local hosted Neo4j Browser web application on your machine with a simple one-line command.
+Also it is easy to run a specific version of Neo4j Browser, download the preffered version from link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[Maven].
+
+Neo4j Browser deployed web application::
+If you want to have specific network configurations within you network and host Neo4j Browser for quick access within your organization, you can host Neo4j Browser with a HTTP web server.
+This allows for full control of configuring the HTTP web server as needed, for example you way want to add a layer of access control to access the Neo4j Browser web application and configure the HTTP web server connection to use TLS.
+
+Extend or improve the source code::
+Neo4j Browser is open source (GNU General Public License v3.0) and the source code can be found at link:https://github.com/neo4j/neo4j-browser[_https://github.com/neo4j/neo4j-browser_].
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:149]:
```suggestion
Neo4j Browser is open source (GNU General Public License v3.0), and the source code can be found at link:https://github.com/neo4j/neo4j-browser[_https://github.com/neo4j/neo4j-browser_].
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 46]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+If you want to extend or contribute to the project you can alter the code.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:150]:
```suggestion
If you want to extend or contribute to the project, you can alter the code.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 47]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+This allows you to run an altered version of Neo4j Browser.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:151]:
```suggestion
This allows you to run a modified version of Neo4j Browser.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 48]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
++
+[source, shell]
+----
+npx http-server -p 8080 neo4j-browser/
+----
++
+For more information, see link:https://www.npmjs.com/package/http-server[_https://www.npmjs.com/package/http-server_].
+
+Node.js -- `serve`::
+A simple HTTP server that is ideal for serving a static site, single page application or just a static file.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:187]:
```suggestion
A simple HTTP server is ideal for serving a static site, single page application, or just a static file.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 49]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+
+Apache HTTP Server::
+For more information, see link:https://httpd.apache.org/[_https://httpd.apache.org/_].
+
+Internet Information Services (IIS)::
+For more information, see link:https://www.iis.net/[_https://www.iis.net/_].
+====
+
+
+[[pre-build-neo4j-browser]]
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:212]:
```suggestion
[[pre-built-neo4j-browser]]
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 50]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+=== Pre-build Neo4j Browser
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:213]:
```suggestion
=== Pre-built Neo4j Browser
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 51]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+
+You can download Neo4j Browser from Maven (link:https://search.maven.org/artifact/org.neo4j.client/neo4j-browser[_https://search.maven.org/artifact/org.neo4j.client/neo4j-browser_]) and serve the web application with an HTTP server.
+
+
+.Get pre-build Neo4j Browser and run an HTTP web server
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:218]:
```suggestion
.Get pre-built Neo4j Browser and run an HTTP web server
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 52]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
++
+[source, shell, subs="attributes+"]
+----
+unzip neo4j-browser-{neo4j-browser-version}.jar
+----
++
+. **Host Neo4j Browser with an HTTP server.**
++
+Start a web server to serve the content.
+In this example you start a web server with Python 3.
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:240]:
```suggestion
In this example, you start a web server with Python 3.
```

File: docs/modules/ROOT/pages/deployment-modes.adoc [hunk 53]
```diff
@@ -54,34 +94,240 @@ image:browser-with-desktop.png[]
...
+[source, shell]
+----
+python -m http.server --directory browser/ 8080
+----
++
+Point your web browser to link:http://localhost:8080[_http://localhost:8080_].
++
+. **Test the connection to a Neo4j database instance.**
++
+If you have the Neo4j tarball you can start the database instance with the command (assuming that the current working directory is _<neo4j-home>_):
```
Comment (renetapopova) [inline docs/modules/ROOT/pages/deployment-modes.adoc:251]:
```suggestion
If you have the Neo4j tarball, you can start the database instance with the command (assuming that the current working directory is _<neo4j-home>_):
```

**Review by renetapopova — CHANGES_REQUESTED** [review]
Looks much better. Some minor suggestions and typos.

**Review by renetapopova — APPROVED** [review]
LGTM, thanks @martin-neotech

**Pull request merged**